src/H5Lint.c

```diff
@@ -596,8 +596,10 @@ H5L__link_cb(H5G_loc_t *grp_loc /*in*/, const char *name, const H5O_link_t H5_AT
 
     /* Set the link's name correctly */
     /* Casting away const OK -QAK */
-    udata->lnk->name = (char *)name;
-
+    udata->lnk->name = strdup(name);
+    if (NULL == udata->lnk->name) {
+        HGOTO_ERROR(H5E_LINK, H5E_CANTINIT, FAIL, "out of memory")
+    }
```

Comment:
Since these changes are in callbacks, it sort of seems like doing a strdup each time probably wouldn't be optimal. This seems like the kind of thing that needs to be fixed somewhere higher up where we correctly differentiate between when the link name should be const and non-const.

If we do go with this change though, the strdup calls should be changed to HDstrdup, or even better, HDstrndup.

Reply:
HDstrndup() is not available in H5private.h. 
strndup() is not available for MS VS C compiler. 
Thus, I replaced strdup() with HDstrdup().

Reply:
> Since these changes are in callbacks, it sort of seems like doing a strdup each time probably wouldn't be optimal. This seems like the kind of thing that needs to be fixed somewhere higher up where we correctly differentiate between when the link name should be const and non-const.

Agree